Go2.html

```diff
@@ -92,25 +92,98 @@
+            stoneRight[i + 342] = 3;
+            stoneTop[i * 19] = 3;
+            stoneBottom[(i * 19) + 18] = 3; //value of 3 indicates both black and white
+        };
+        
+        let whoseTurn = 2; //1 = Black 2 = White
         for (let i = 0; i < 361; i++){
             const stoneElement = document.getElementById(stoneIDArray[i]);
             stoneElement.addEventListener("mouseover", function(stone) {
-                if (stoneColourArray[i] < 1) {
+                if (stoneColourArray[i] < 1 && whoseTurn == 1) {
                     stone.target.style.opacity = "0.3";
-                }
+                    stone.target.style.backgroundImage = "radial-gradient(rgb(85, 85, 85),black)";
+                } else if (stoneColourArray[i] < 1 && whoseTurn == 2) {
+                    stone.target.style.opacity = "0.3";
+                    stone.target.style.backgroundImage = "radial-gradient(rgb(185, 185, 185),rgb(255, 255, 255))";
+                } 
             });
+
             stoneElement.addEventListener("mouseout", function(stone) {
                 if (stoneColourArray[i] < 1) {
                     stone.target.style.opacity = "0";
                 }
             });
+
             stoneElement.addEventListener("click", function(stone) {
                 if (stoneColourArray[i] < 1) {
                     stone.target.style.opacity = "1";
-                    stoneColourArray[i] = 1;
-                }
+                    if (whoseTurn == 1){
+                        stoneColourArray[i] = 1;
+                        whoseTurn = 2;
+                        stone.target.style.backgroundImage = "radial-gradient(rgb(85, 85, 85),black)";
+                        stoneLeft[i + 19] = 1;
+                        stoneRight[i - 19] = 1;
+                        stoneTop[i + 1] = 1;
+                        stoneBottom[i - 1] = 1;
+                    } else if (whoseTurn == 2){
+                        stoneColourArray[i] = 2;
+                        whoseTurn = 1;
+                        stone.target.style.backgroundImage = "radial-gradient(rgb(185, 185, 185),rgb(255, 255, 255))";
+                        stoneLeft[i + 19] = 2;
+                        stoneRight[i - 19] = 2;
+                        stoneTop[i + 1] = 2;
+                        stoneBottom[i - 1] = 2;
+                    };
+                };
             });
-        }
+        };
+                for (let i = 0; i < 361; i++) {
+                function changeStoneSurrounded(stoneSurroudned){
+                    if (stoneLeft[i] == (1 || 3) && stoneRight[i] == (1 || 3) && stoneTop[i] == (1 || 3) && stoneBottom[i] == (1 || 3)) {
+                    stoneSurroudned[i] = 1;
+                    
+                console.log(stoneSurroudned[i])
+                }
+            } 
+        };
+
+        let illegalMove = []; //0 = all can place stone 1: black can't place stone 2: white can't place stone
+        for (let i = 0; i < 361; i++) {
```

Comment:
So if we want to create an array with just 0's, there's a more efficient way https://stackoverflow.com/questions/1295584/most-efficient-way-to-create-a-zero-filled-javascript-array. In this case we can do 
```suggestion
        let illegalMove = new Array(360).fill(0);
```